src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -46,6 +47,20 @@ function listenForNavigationEvents(
   currentHistoryIndex = history.state?._index ?? 0;
 }
 
+async function enableNavigationInterception(uriInDotNet?: string) {
+  setHasInteractiveRouter(true);
+  if (uriInDotNet && location.href !== uriInDotNet) {
```

Comment:
In what scenarios does this happen?

Reply:
Here's an example:
1. The user is currently on a page with an interactive router
2. The user navigates to a page without an interactive router, disposing the router and disabling navigation interception
3. The user continues to click around, all while .NET is not getting notified of location changes (because no interactive router is present)
4. The user navigates back to a page with an interactive router. Navigation interception gets enabled, but the itneractive router didn't get a chance to detect the navigation that just happened, so the interactive .NET `NavigationManager` still has the wrong URL and will therefore render content for the wrong page (or 404).